packages/eslint-plugin-office-addins/src/rules/test-for-null-using-isNullObject.ts

```diff
@@ -0,0 +1,155 @@
+    },
+    schema: [],
+    fixable: <"code" | "whitespace"> "code",
+  },
+  create: function (context: any) {
+    function isConditionalTestExpression(node: TSESTree.Identifier): boolean {
+      return (
+        node.parent != undefined &&
+        (node.parent.type === AST_NODE_TYPES.IfStatement ||
+          node.parent.type === AST_NODE_TYPES.WhileStatement ||
+          node.parent.type === AST_NODE_TYPES.DoWhileStatement ||
+          node.parent.type === AST_NODE_TYPES.ForStatement ||
+          node.parent.type === AST_NODE_TYPES.ConditionalExpression) &&
+        node === node.parent.test
+      );
+    }
+
+    function isInUnaryNullTest(node: TSESTree.Identifier): boolean {
+      return (
+        node.parent != undefined &&
+        node.parent.type === AST_NODE_TYPES.UnaryExpression &&
+        node.parent.operator === "!" &&
+        node.parent.argument === node
+      );
+    }
+
+    function isInBinaryNullTest(node: TSESTree.Identifier): boolean {
+      return (
+        node.parent != undefined &&
+        node.parent.type === AST_NODE_TYPES.BinaryExpression &&
+        ((node.parent.left === node &&
+          node.parent.right.type === AST_NODE_TYPES.Literal &&
+          node.parent.right.raw === "null") ||
+          (node.parent.right === node &&
+            node.parent.left.type === AST_NODE_TYPES.Literal &&
+            node.parent.left.raw === "null"))
+      );
+    }
+
+    function isInNullTest(node: TSESTree.Identifier): boolean {
+      return (
+        isConditionalTestExpression(node) ||
+        node.parent?.type === AST_NODE_TYPES.LogicalExpression ||
+        isInUnaryNullTest(node) ||
+        isInBinaryNullTest(node)
+      );
+    }
+
+    function isNullObjectCall(node: TSESTree.Node): boolean {
+      if (
+        node.type === AST_NODE_TYPES.CallExpression &&
+        node.callee.type === AST_NODE_TYPES.MemberExpression &&
+        node.callee.property.type === AST_NODE_TYPES.Identifier &&
+        node.callee.property.name.endsWith("OrNullObject")
+      ) {
+        return true;
+      }
+      return false;
+    }
+
+    function isNullObjectNode(node: TSESTree.Node | undefined): boolean {
+      if (
+        node &&
+        ((node.type === AST_NODE_TYPES.VariableDeclarator &&
+          node.init &&
+          isNullObjectCall(node.init) &&
+          node.id.type === AST_NODE_TYPES.Identifier) ||
+          (node.type === AST_NODE_TYPES.AssignmentExpression &&
+            isNullObjectCall(node.right) &&
+            node.left.type === AST_NODE_TYPES.Identifier))
+      ) {
+        return true;
+      }
+      return false;
+    }
+
+    function findNullObjectNullTests(scope: any): void {
+      const variables = scope.variables;
+      const childScopes = scope.childScopes;
+      let i, l;
+
+      for (i = 0, l = variables.length; i < l; i++) {
```

Comment:
Can you change the `l` variable? It doesn't even seem needed (It only has the lenght value) and makes the code more confusing

Reply:
Copied from the style used by eslint.  Don't know if there is a technical cost to reading the member variable repeatedly, but I can certainly change it.

Reply:
I think it shouldn't have any difference in performance, as typescript (or js) should optimize this call underneath.